src/js/pages/catalog/PackagesTab.js

```diff
@@ -284,26 +350,47 @@ class PackagesTab extends mixin(StoreMixin) {
         const splitPackages = packages.getSelectedAndNonSelectedPackages();
 
         let communityPackages = splitPackages.nonSelectedPackages;
-        const selectedPackages = splitPackages.selectedPackages;
+        let selectedPackages = splitPackages.selectedPackages;
 
         if (state.searchString) {
-          communityPackages = packages.filterItemsByText(state.searchString);
+          selectedPackages = selectedPackages.filterItemsByText(
+            state.searchString
+          );
+          communityPackages = communityPackages.filterItemsByText(
+            state.searchString
+          );
         }
 
         content = (
           <div className="container">
             <div className="pod flush-horizontal flush-top">
-              <FieldAutofocus>
-                <FilterInputText
-                  className="flex-grow"
-                  placeholder="Search catalog"
-                  searchString={state.searchString}
-                  handleFilterChange={this.handleSearchStringChange}
-                />
-              </FieldAutofocus>
+              <FilterBar>
+                <FieldAutofocus>
+                  <FilterInputText
+                    className="flush-bottom"
+                    placeholder="Search catalog"
+                    searchString={state.searchString}
+                    handleFilterChange={this.handleSearchStringChange}
+                  />
+                </FieldAutofocus>
+                {this.getFilterButtons(
+                  selectedPackages.list.length,
+                  communityPackages.list.length
+                )}
+              </FilterBar>
             </div>
-            {this.getCertifiedPackagesGrid(selectedPackages)}
-            {this.getCommunityPackagesGrid(communityPackages)}
+            {state.searchFilter !== COMMUNITY
+              ? this.getCertifiedPackagesGrid(
```

Comment:
I debated having only one function for packages grid, but in the end I decided against this.